lib/ceedling/configurator.rb

```diff
@@ -67,40 +67,44 @@ def reset_defaults(config)
 
   # The default values defined in defaults.rb (eg. DEFAULT_TOOLS_TEST) are populated
   # into @param config
-  def populate_defaults(config)
-    new_config = DEFAULT_CEEDLING_CONFIG.deep_clone
-    new_config.deep_merge!(config)
-    config.replace(new_config)
-
-    @configurator_builder.populate_defaults( config, DEFAULT_TOOLS_TEST )
-    @configurator_builder.populate_defaults( config, DEFAULT_TOOLS_TEST_PREPROCESSORS ) if (config[:project][:use_test_preprocessor])
-    @configurator_builder.populate_defaults( config, DEFAULT_TOOLS_TEST_DEPENDENCIES )  if (config[:project][:use_deep_dependencies])
-
-    @configurator_builder.populate_defaults( config, DEFAULT_TOOLS_RELEASE )              if (config[:project][:release_build])
-    @configurator_builder.populate_defaults( config, DEFAULT_TOOLS_RELEASE_ASSEMBLER )    if (config[:project][:release_build] and config[:release_build][:use_assembly])
-    @configurator_builder.populate_defaults( config, DEFAULT_TOOLS_RELEASE_DEPENDENCIES ) if (config[:project][:release_build] and config[:project][:use_deep_dependencies])
-  end
+  def merge_ceedling_defaults(config, default_config)
+
+    # Merge ceedling default config with default tools
+    default_config.replace( DEFAULT_CEEDLING_CONFIG.deep_clone )
+    default_config.deep_merge( DEFAULT_TOOLS_TEST )
```

Comment:
```suggestion
    default_config.deep_merge( DEFAULT_TOOLS_TEST.deep_clone )
```
and similarly for the following lines.